streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregate.java

```diff
@@ -148,7 +153,7 @@ public void processInOrder(final K key, final V value, final long timestamp) {
             boolean rightWinAlreadyCreated = false;
 
             // keep the left type window closest to the record
-            Window latestLeftTypeWindow = null;
+            KeyValue<Windowed<K>, ValueAndTimestamp<Agg>> latestLeftTypeWindow = null;
             try (
                     final KeyValueIterator<Windowed<K>, ValueAndTimestamp<Agg>> iterator = windowStore.fetch(
```

Comment:
Fetch having negative bounds doesn't throw any errors or cause any issues, is there a different reason to make sure the bounds aren't negative? Since we don't store windows with a negative start time it shouldn't return anything we don't expect

Reply:
Are you sure it's actually returning something? Have you tested it with a rocksdb store or just with the in-memory store? I think the in-memory store would handle this fine since it never serializes the key/timestamps, but if you have a rocksdb store (or a caching layer) then the range query works by looking up any data between the serialized bounds. Unfortunately a negative long is lexicographically greater than a positive long when serialized to bytes. The "negative" is encoded as a leading 1 -- which means the lower bound ends up being "larger" than the upper bound. I assume that would result in no data being returned, but I'm not actually 100% sure what would happen

Reply:
Ah yeah the different types of stores is a good point, I haven't tested with all of them. I changed it to do 0 or the positive fetch start